### PR TITLE
Add operation definition profiles

### PR DESCRIPTION
**Pull request by bkaney**
This PR has the profiles for operation definitions, namely:

* Artifact Endpoint Configurable Operation
* Canonical Operation
* Data Configurable Operation
* Manifestable Operation
* Pagable Operation 
* VersionBindable Operation

And applies them to data-requirement and package operations.

TODO: @brynrhodes  
- Are "CRMI ValueSet Expand", and "CRMI Get Package" operations deprecated?
- I am not familiar with "release" operation, so I didn't update that in this PR

### CODE REVIEW

File: input/fsh/operation-profiles/versionable-operation-profile.fsh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Profile: VersionableOperation
```
Comment (bkaney) [inline input/fsh/operation-profiles/versionable-operation-profile.fsh:1]:
```suggestion
Profile: VersionableBindableOperation
```

File: input/fsh/operation-profiles/versionable-operation-profile.fsh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+Id: crmi-versionable-operation
```
Comment (bkaney) [inline input/fsh/operation-profiles/versionable-operation-profile.fsh:2]:
```suggestion
Id: crmi-version-bindable-operation
```

File: input/fsh/operation-profiles/versionable-operation-profile.fsh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+Parent: OperationDefinition
+Title: "CRMI Operation Profile: Versionable"
```
Comment (bkaney) [inline input/fsh/operation-profiles/versionable-operation-profile.fsh:4]:
```suggestion
Title: "CRMI Operation Profile: Version Bindable"
```

File: input/pages/operations.md [hunk 4]
```diff
@@ -1,174 +1,31 @@
 {: #operations}
 
+OperationDefinitions that involve canonical resources may conform to one or more
+of the following profiles. This allows conventionalize operation paramater
+names, types, and processing semantics.
+
+| **Operation Profile**               | **Purpose**                                                                 |
+|-------------------------------------|-----------------------------------------------------------------------------|
+| [Canonical][1]                      | Defines parameters used to identify canonical resource                      |
+| [Versionable][2]                    | Defines parameters used to specify missing versions in canonical references |
```
Comment (bkaney) [inline input/pages/operations.md:10]:
```suggestion
| [Version Bindable][2]               | Defines parameters used to specify missing versions in canonical references |
```

File: input/pages/operations.md [hunk 5]
```diff
@@ -1,174 +1,31 @@
...
+| [Manifestable][3]                   | Defines parameters used to provide a version manifest                       |
+| [Pagable][4]                        | Defines parameters for an operation with pagable options                    |
+| [Artifact Endpoint Configurable][5] | Defines parameters to provide endpoints to resolve canonical artifacts      |
+| [Data Configurable][6]              | Defines parameters to specify how to resolve data requirements              |
+
+[1]: StructureDefinition-crmi-canonical-operation.html
+[2]: StructureDefinition-crmi-versionable-operation.html
```
Comment (bkaney) [inline input/pages/operations.md:17]:
```suggestion
[2]: StructureDefinition-crmi-version-bindable-operation.html
```

File: input/fsh/metadata.fsh [hunk 6]
```diff
@@ -6,6 +6,23 @@ RuleSet: DefinitionMetadata
...
+RuleSet: DataConfigurableOperationProfile
+* meta.profile[+] = Canonical(DataConfigurableOperation)
+
+RuleSet: ManifestableOperationProfile 
+* meta.profile[+] = Canonical(ManifestableOperation)
+
+RuleSet: PageableOperationProfile 
+* meta.profile[+] = Canonical(PageableOperation)
+
+RuleSet: VersionableOperationProfile
```
Comment (bkaney) [inline input/fsh/metadata.fsh:27]:
```suggestion
RuleSet: VersionBindableOperationProfile
```

File: input/fsh/metadata.fsh [hunk 7]
```diff
@@ -6,6 +6,23 @@ RuleSet: DefinitionMetadata
...
+* meta.profile[+] = Canonical(VersionableOperation)
```
Comment (bkaney) [inline input/fsh/metadata.fsh:28]:
```suggestion
* meta.profile[+] = Canonical(VersionBindableOperation)
```

File: input/fsh/operation-definitions/data-requirements-operation.fsh [hunk 8]
```diff
@@ -3,6 +3,9 @@ InstanceOf: OperationDefinition
 Title: "CRMI Data Requirements Operation"
 Usage: #definition
 * insert DefinitionMetadata
+* insert CanonicalOperationProfile
+* insert VersionableOperationProfile
```
Comment (bkaney) [inline input/fsh/operation-definitions/data-requirements-operation.fsh:7]:
```suggestion
* insert VersionBindableOperationProfile 
```

File: input/fsh/operation-definitions/package-operation.fsh [hunk 9]
```diff
@@ -3,6 +3,9 @@ InstanceOf: OperationDefinition
 Title: "CRMI Package Operation"
 Usage: #definition
 * insert DefinitionMetadata
+* insert CanonicalOperationProfile
+* insert VersionableOperationProfile
```
Comment (bkaney) [inline input/fsh/operation-definitions/package-operation.fsh:7]:
```suggestion
* insert VersionBindableOperationProfile 
```

File: input/fsh/operation-profiles/_shared.fsh [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+* $fhir-types#ResearchDefinition
+* $fhir-types#ResearchElementDefinition
+* $fhir-types#RiskEvidenceSynthesis
+* $fhir-types#SearchParameter
+* $fhir-types#StructureDefinition
+* $fhir-types#StructureMap
+* $fhir-types#TerminologyCapabilities
+* $fhir-types#TestScript
+* $fhir-types#ValueSet
+// also the "additional" canonical types
```
Comment (brynrhodes) [inline input/fsh/operation-profiles/_shared.fsh:33]: We should add a "ArtifactResourceTypes" that includes the CanonicalResourceTypes and the non-canonical definitional resource types

File: input/fsh/operation-profiles/canonical-operation-profile.fsh [hunk 11]
```diff
@@ -0,0 +1,56 @@
+Profile: CanonicalOperation
```
Comment (brynrhodes) [inline input/fsh/operation-profiles/canonical-operation-profile.fsh:1]: Rename this profile to ArtifactOperation

File: input/fsh/operation-profiles/versionable-operation-profile.fsh [hunk 12]
```diff
@@ -0,0 +1,39 @@
+Profile: VersionableBindableOperation
```
Comment (brynrhodes) [inline input/fsh/operation-profiles/versionable-operation-profile.fsh:1]: This becomes CanonicalVersionBindable operation, also create a new ArtifactVersionBindable operation that defines artifactVersion, checkArtifactVersion, and forceArtifactVersion parameters

File: input/fsh/operation-definitions/data-requirements-operation.fsh [hunk 13]
```diff
@@ -147,12 +150,12 @@ format is the same as a canonical URL: [url]|[version] - e.g. http://loinc.org|2
 Note that this is a generalization of the `check-system-version` parameter to the $expand operation to 
 apply to any canonical resource, including code systems.
 """
-* parameter[=].type = #canonical
+* parameter[=].type = #boolean
```
Comment (brynrhodes) [inline input/fsh/operation-definitions/data-requirements-operation.fsh:154]: This should be uri right, not boolean?

File: input/fsh/operation-definitions/data-requirements-operation.fsh [hunk 14]
```diff
@@ -147,12 +150,12 @@ format is the same as a canonical URL: [url]|[version] - e.g. http://loinc.org|2
...
 
-* parameter[+].name = #forceCanonicalVersion
+* parameter[+].name = #forceArtifactlVersion
```
Comment (brynrhodes) [inline input/fsh/operation-definitions/data-requirements-operation.fsh:155]: forceArtfactVersion, not forceArtifactlVersion

File: input/fsh/operation-definitions/data-requirements-operation.fsh [hunk 15]
```diff
@@ -147,12 +150,12 @@ format is the same as a canonical URL: [url]|[version] - e.g. http://loinc.org|2
...
 * parameter[=].use = #in
 * parameter[=].min = 0
-* parameter[=].max = "*"
+* parameter[=].max = "1"
```
Comment (brynrhodes) [inline input/fsh/operation-definitions/data-requirements-operation.fsh:158]: Needs to be * still, you can do this any number of times

File: input/fsh/operation-definitions/data-requirements-operation.fsh [hunk 16]
```diff
@@ -168,7 +171,7 @@ explicitly be represented in the expansion parameters.
 Note that this is a generalization of the `force-system-version` parameter to the $expand operation 
 to apply to any canonical resource, including code systems.
 """
-* parameter[=].type = #canonical
+* parameter[=].type = #boolean
```
Comment (brynrhodes) [inline input/fsh/operation-definitions/data-requirements-operation.fsh:174]: Should be uri

**Review by brynrhodes — CHANGES_REQUESTED** [review]
A couple comments

**Review by brynrhodes — APPROVED** [review]

**Pull request merged**